x/ibc/03-connection/keeper/handshake_test.go

```diff
@@ -75,43 +77,103 @@ func (suite *KeeperTestSuite) TestConnOpenTry() {
+
 			consensusHeight = uint64(suite.chainB.GetContext().BlockHeight())
 		}, false},
 		{"self consensus state not found", func() {
 			clientA, clientB = suite.coordinator.SetupClients(suite.chainA, suite.chainB, clientexported.Tendermint)
 			_, _, err := suite.coordinator.ConnOpenInit(suite.chainA, suite.chainB, clientA, clientB)
 			suite.Require().NoError(err)
 
+			// retrieve client state of chainA to pass as counterpartyClient
+			counterpartyClient, found = suite.chainA.App.IBCKeeper.ClientKeeper.GetClientState(suite.chainA.GetContext(), clientA)
+			suite.Require().True(found)
+
 			consensusHeight = 1
 		}, false},
 		{"counterparty versions is empty", func() {
 			clientA, clientB = suite.coordinator.SetupClients(suite.chainA, suite.chainB, clientexported.Tendermint)
 			_, _, err := suite.coordinator.ConnOpenInit(suite.chainA, suite.chainB, clientA, clientB)
 			suite.Require().NoError(err)
 
+			// retrieve client state of chainA to pass as counterpartyClient
+			counterpartyClient, found = suite.chainA.App.IBCKeeper.ClientKeeper.GetClientState(suite.chainA.GetContext(), clientA)
+			suite.Require().True(found)
+
 			versions = nil
 		}, false},
 		{"counterparty versions don't have a match", func() {
 			clientA, clientB = suite.coordinator.SetupClients(suite.chainA, suite.chainB, clientexported.Tendermint)
 			_, _, err := suite.coordinator.ConnOpenInit(suite.chainA, suite.chainB, clientA, clientB)
 			suite.Require().NoError(err)
 
+			// retrieve client state of chainA to pass as counterpartyClient
+			counterpartyClient, found = suite.chainA.App.IBCKeeper.ClientKeeper.GetClientState(suite.chainA.GetContext(), clientA)
+			suite.Require().True(found)
+
 			version, err := types.NewVersion("0.0", nil).Encode()
 			suite.Require().NoError(err)
 			versions = []string{version}
 		}, false},
 		{"connection state verification failed", func() {
 			clientA, clientB = suite.coordinator.SetupClients(suite.chainA, suite.chainB, clientexported.Tendermint)
 			// chainA connection not created
+
+			// retrieve client state of chainA to pass as counterpartyClient
+			counterpartyClient, found = suite.chainA.App.IBCKeeper.ClientKeeper.GetClientState(suite.chainA.GetContext(), clientA)
+			suite.Require().True(found)
+		}, false},
+		{"client state verification failed", func() {
+			clientA, clientB = suite.coordinator.SetupClients(suite.chainA, suite.chainB, clientexported.Tendermint)
+			_, _, err := suite.coordinator.ConnOpenInit(suite.chainA, suite.chainB, clientA, clientB)
+			suite.Require().NoError(err)
+
+			// retrieve client state of chainA to pass as counterpartyClient
+			counterpartyClient, found = suite.chainA.App.IBCKeeper.ClientKeeper.GetClientState(suite.chainA.GetContext(), clientA)
+			suite.Require().True(found)
+
+			// modify counterparty client so it still passes validate but fails proof verification
+			// Set an invalid client of chainA on chainB
+			tmClient, ok := counterpartyClient.(*ibctmtypes.ClientState)
```

Comment:
does this not need to be set in the keeper again?

Reply:
Whoops, godoc is wrong from copy pasting. I modify but DO NOT set in store so that the proof verification fails while ValidateClient called earlier, passes.

Updated comment to show this